main/lsp/requests/hover.cc

```diff
@@ -48,17 +62,39 @@ LSPResult LSPLoop::handleTextDocumentHover(unique_ptr<core::GlobalState> gs, con
         }
 
         auto resp = move(queryResponses[0]);
+
+        optional<string> documentation = nullopt;
+        if (resp->isConstant() || resp->isDefinition()) {
+            auto origins = resp->getTypeAndOrigins().origins;
+            if (!origins.empty()) {
+                auto loc = origins[0];
+                if (loc.exists()) {
+                    documentation = findDocumentation(loc.file().data(*gs).source(), loc.beginPos());
+                }
+            }
+        }
+
         if (auto sendResp = resp->isSend()) {
             auto retType = sendResp->dispatchResult->returnType;
+            // TODO main.method is <none> when calling .new on a class.
```

Comment:
You can remove this TODO, right? We discovered in our pairing session that this is not always true, and you find docs for `new` in some scenarios.

Reply:
b12eeff7b